### ISSUE TITLE
NPE in OptionsConfigurationBlock

### ISSUE THREAD

**Issue by tivervac** [eclipse/xtext-eclipse#191]
I am getting this error clicking on the "Configure Project Specific Settings..." in the Task Tags view in the Preferences of my app.
What I think is happening is the following:

The `TaskTagPreferencePage` [creates](https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/tasks/preferences/TaskTagPreferencePage.java#L52) a `TaskTagConfigurationBlock` using the new keyword instead of injecting it.

When we then call `OptionsConfigurationBlock.hasProjectSpecificOptions(IProject)` the `preferenceStoreAccessImpl` [has to be accessed](https://github.com/eclipse/xtext-eclipse/blob/master/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/preferences/OptionsConfigurationBlock.java#L237), but it is null. It is null because it should've been injected, but injection failed as `TaskTagConfigurationBlock` was not injected itself.

My task tags view is created in the plugin.xml of my app in the following way:
```
<page
	category="com.foo.bar"
   	class="com.foo.bar.ExecutableExtensionFactory:org.eclipse.xtext.ui.tasks.preferences.TaskTagPreferencePage"
	id="com.foo.bar.task.preferencePage"
	name="Task Tags">
	<keywordReference id="com.foo.bar.ui.keyword_Foobar"/>
</page>
```

**Comment by cdietrich** [comment]
Another ugly detail: it uses PreferenceStoreAccessImpl insteadof IPreferenceStoreAccess

**Comment by cdietrich** [comment]
fixed in 2.12
